Tests/DataStreamTests.swift

```diff
@@ -0,0 +1,749 @@
+        var response: HTTPURLResponse?
+        var complete: DataStreamRequest.Completion?
+        let expect = expectation(description: "stream complete")
+
+        // When
+        AF.streamRequest(URLRequest.makeHTTPBinRequest(path: "bytes/50"), automaticallyCancelOnStreamError: true)
+            .responseStreamDecodable(of: HTTPBinResponse.self) { stream in
+                switch stream.event {
+                case let .complete(completion):
+                    complete = completion
+                    response = completion.response
+                    expect.fulfill()
+                default: break
+                }
+            }
+
+        waitForExpectations(timeout: timeout)
+
+        // Then
+        XCTAssertEqual(response?.statusCode, 200)
+        XCTAssertTrue(complete?.error?.isExplicitlyCancelledError == true)
+    }
+
+    func testThatDataStreamIsAutomaticallyCanceledOnStreamClosureError() {
+        // Given
+        enum LocalError: Error { case failed }
+
+        var response: HTTPURLResponse?
+        var complete: DataStreamRequest.Completion?
+        let expect = expectation(description: "stream complete")
+
+        // When
+        AF.streamRequest(URLRequest.makeHTTPBinRequest(path: "bytes/50"))
+            .responseStream { stream in
+                switch stream.event {
+                case .stream: throw LocalError.failed
+                case let .complete(completion):
+                    complete = completion
+                    response = completion.response
+                    expect.fulfill()
+                }
+            }
+
+        waitForExpectations(timeout: timeout)
+
+        // Then
+        XCTAssertEqual(response?.statusCode, 200)
+        XCTAssertTrue(complete?.error?.isExplicitlyCancelledError == false)
+    }
+
+    func testThatDataStreamCanBeCancelledInClosure() {
+        // Given
+        let expectedSize = 1000
+        var error: AFError?
+        let expect = expectation(description: "stream should complete")
+
+        // When
+        AF.streamRequest("https://httpbin.org/bytes/\(expectedSize)").responseStream { stream in
+            switch stream.event {
+            case .stream:
+                stream.cancel()
+            case .complete:
+                error = stream.completion?.error
+                expect.fulfill()
+            }
+        }
+
+        waitForExpectations(timeout: timeout)
+
+        // Then
+        XCTAssertTrue(error?.isExplicitlyCancelledError == true)
+    }
+
+    func testThatDataStreamCanBeCancelledByToken() {
+        // Given
+        let expectedSize = 1000
+        var error: AFError?
+        let expect = expectation(description: "stream should complete")
+
+        // When
+        AF.streamRequest("https://httpbin.org/bytes/\(expectedSize)").responseStream { stream in
+            switch stream.event {
+            case .stream:
+                stream.token.cancel()
```

Comment:
Is there really a need to expose the `token` if you're exposing the `cancel` API on `stream`?

Reply:
I imagined the `token` could be saved off separately so users could pass it around. I'm not sure how likely that is but I thought the flexibility might be useful.